src/bridging/types.ts

```diff
@@ -0,0 +1,177 @@
+import { latest as latestAppData } from '@cowprotocol/app-data'
+import { ethers } from 'ethers'
+import { ChainInfo, ChainId } from '../common'
+import { TokenInfo } from '../common/types/tokens'
+import { Address, OrderKind } from '../order-book'
+import { EvmCall } from '../common/types/ethereum'
+
+export interface BridgeProviderInfo {
+  name: string
+  logoUrl: string
+}
+
+export interface WithSellToken {
+  sellTokenChainId: ChainId
+  sellTokenAddress: Address
+  sellTokenDecimals: number
+}
+
+export interface WithBuyToken {
+  buyTokenChainId: ChainId
+  buyTokenAddress: Address
+  buyTokenDecimals: number
+}
+
+export interface GetBuyTokensParams extends Partial<WithSellToken> {
+  targetChainId: ChainId
+}
+
+/**
+ * Parameters for getting a bridge quote
+ */
+export interface QuoteBridgeRequest extends WithSellToken, WithBuyToken {
+  type: OrderKind.SELL // We make it explicit that only SELL is supported for now
+  amount: string
+  recipient: string
+}
+
+export interface QuoteFee {
+  type: 'fixed' | 'percent'
+  value: number
+}
+
+export interface BridgeQuoteResult {
+  fee: QuoteFee
+  slippageBps: number
+  buyAmount: string
+  fillTimeInSeconds?: number
+}
+
+export interface BridgeHook {
+  postHook: latestAppData.CoWHook
+  recipient: string
+}
+
+export enum BridgeStatus {
+  NOT_INITIATED = 'not_initiated',
+  INITIATED = 'initiated',
+  EXECUTED = 'executed',
+  FAILED = 'failed',
+  EXPIRED = 'expired',
+}
+
+export interface BridgeStatusResult {
+  status: BridgeStatus
+  fillTimeInSeconds?: number
+}
+
+/**
+ * A bridge deposit. It includes the sell amount and the minimum buy amount.
+ *
+ * It models the minimum information needed to initiate a bridge and that it can also be extracted from the cow hook.
+ *
+ */
+export interface BridgeDeposit extends Omit<QuoteBridgeRequest, 'amount'> {
+  provider: BridgeProviderInfo
+
+  sellTokenAmount: string
+  minBuyAmount: string
+}
+
+export interface BridgeProvider<Q extends BridgeQuoteResult> {
+  info: BridgeProviderInfo
+
+  /**
+   * Get basic supported chains
+   */
+  getNetworks(): Promise<ChainInfo[]>
+
+  /**
+   * Get supported tokens for a chain
+   */
+  getBuyTokens(chainId: GetBuyTokensParams): Promise<TokenInfo[]>
+
+  /**
+   * Get intermediate tokens given a quote request.
+   *
+   * An intermediate token, is a token in the source chain, that could be used to bridge the tokens to the destination chain.
+   * This method returns a sorted list of tokens, they are sorted by priority, so first tokens are more likely to be more liquid.
+   *
+   * @param request - The quote request
+   */
+  getIntermediateTokens(request: QuoteBridgeRequest): Promise<string[]>
```

Comment:
Should it be a part of `getQuote()` response?
More likely we will always use them together

Reply:
The idea is to give more freedom to the SDK to implement the strategies.

Here the sellToken is unknown, so I can't ask the provider for a quote for an unknown sellToken. The idea is to be able to ask providers for these tokens, then be able to get an estimate in CoW Protocol of how many intermediate tokens we can get, and with that we can do the actual getQuote.